### ISSUE TITLE
Failed authentication not reported in the logs

### ISSUE THREAD

**Issue by mdallaire** [emqx/emqx#11932]
### What happened?

I am using the built-in database for authentication and console logs set to warning (default for the Helm chart).

While troubleshooting an issue with a mqtt client I realized that a login failure (wrong username or password) does not get logged to the global log.

I am able to see the login failure in a trace log.

### What did you expect to happen?

I expect a login failure to appear in the log as stated in the [documentation](https://www.emqx.io/docs/en/latest/observability/log.html#log-level)

### How can we reproduce it (as minimally and precisely as possible)?

Deploy the helm chart with the following values.
```
  values:
    emqxConfig:
      EMQX_ALLOW_ANONYMOUS: "false"
      EMQX_AUTHENTICATION__1__MECHANISM: password_based
      EMQX_AUTHENTICATION__1__BACKEND: built_in_database
      EMQX_AUTHENTICATION__1__USER_ID_TYPE: username
      EMQX_AUTHENTICATION__1__PASSWORD_HASH_ALGORITHM: {"name":"plain", "salt_position":"disable"}
      EMQX_AUTHENTICATION__1__ENABLE: true
      EMQX_LOG__CONSOLE__LEVEL: warning
      EMQX_LOG__CONSOLE__ENABLE: true
      EMQX_LOG__CONSOLE__FORMATTER: text
      EMQX_LOG__CONSOLE__TIME_OFFSET: system
```
Create a user in the dashboard and try to login with a wrong password while monitoring the logs `kubectl logs statefulsets/emqx -f`

### EMQX version

<details>

```console
sysdescr  : EMQX
version   : 5.3.0
datetime  : 2023-11-12T18:44:50.547090074+00:00
uptime    : 16 minutes, 51 seconds
```

</details>

### OS version

<details>

```console
Kubernetes 1.27.7
OS: Talos v1.5.5
Kernel: 6.1.61-talos
Container-Runtime: containerd://1.6.23
```

</details>

### Log files

[node_dump_20231112_184754.tar.gz](https://github.com/emqx/emqx/files/13328898/node_dump_20231112_184754.tar.gz)

**Comment by mdallaire** [comment]
I am seeing the same issue with 5.3.1.

**Comment by savonarola** [comment]
Not logging auth attempts is a feature of v5 rather than a bug.

Also, for troubleshooting clients, one may also use the tracing feature: https://www.emqx.io/docs/en/latest/observability/tracer.html.
In traces, there is much information about the authentication process.

**Comment by mdallaire** [comment]
Hello! Thanks for the answer.

I am a bit confused as to why this is considered a feature?? Logging failed login attempts is a very much standard behavior with any product making use of authentication.

I was planning to write a parser and scenario to use with the Crowdsec IDS/IPS to detect and block IPs trying to brute force logins. But this is not possible if the only way to get the failed login log is via a manually triggered trace.

I was indeed able to figure out the client issue (which was not auth related in the end) via the traces.

**Comment by zmstone** [comment]
Hi @mdallaire 
This is mainly due to excessive logging.
For brokers exposing the MQTT endpoints to the internet, there could be massive amount of logs.
We are planning to add a throttled-log feature in future release (e.g. to log 1 entry every X minutes), 
when throttled-log feature is in place, we'll perhaps consider adding authentication failure logs to warning level.

**Comment by artiommocrenco** [comment]
@zmstone too much logs? ok, this could be put in DEBUG
in any case, this is crucial for authentication
no information has be hidden from admin about failed authentication attempts, like you are suggesting, to throttle it

**Comment by zmstone** [comment]
Added `authentication_failure` throttled log in this PR: https://github.com/emqx/emqx/pull/12746
Logs are never throttled when the logging level is set at `debug`.
Throttled logs are always traceable even if they are throttled in the standard logging outputs (file | console)

**Comment by artiommocrenco** [comment]
> Added warning level `authentication_failure` throttled log in this PR: https://github.com/emqx/emqx/pull/12746 (will be released in 5.6.0)
> 
> Logs are never throttled when the logging level is set at `debug`.
> 
> Throttled logs are always traceable even if they are throttled in the standard logging outputs (file | console)

this is awesome 🙏
